.github/workflows/publish.yml

```diff
@@ -0,0 +1,81 @@
+name: Publish and deploy
+on: create
+
+env:
+  SERVICE_NAME: ${{ github.event.repository.name }}
+  GOOGLE_PROJECT: broad-dsp-gcr-public
+
+jobs:
+  publish-job:
+    # only publish on new tag
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: v1-${{ runner.os }}-gradle-refs/heads/dev-${{ github.sha }}
+
+      - name: Parse tag
+        id: tag
+        run: echo ::set-output name=tag::$(git describe --tags)
+
+      - name: Publish to Artifactory
+        run: ./gradlew --build-cache :client-resttemplate:artifactoryPublish
+        env:
+          ARTIFACTORY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+          ARTIFACTORY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+          ARTIFACTORY_REPO_KEY: libs-snapshot-local
+
+      - name: Auth to GCR
+        uses: google-github-actions/auth@v0
+        with:
+          credentials_json: ${{ secrets.GCR_PUBLISH_KEY_B64 }}
+      - name: Explicitly auth Docker for GCR
+        run: gcloud auth configure-docker --quiet
+
+      - name: Construct docker image name and tag
+        id: image-name
+        run: echo ::set-output name=name::gcr.io/${GOOGLE_PROJECT}/${SERVICE_NAME}:${{ steps.tag.outputs.tag }}
+
+      - name: Add Google Cloud Profiler to Docker Image
+        run: docker build ./service -t billingprofilemanager:local
+
+      - name: Build image locally with jib
+        run: |
+          ./gradlew --build-cache :service:jibDockerBuild \
+          --image=${{ steps.image-name.outputs.name }} \
+          -Djib.from.image=docker://billingprofilemanager:local \
```

Comment:
I'm wondering about this too, since we don't use this directive when calling `jib` in our workflow. I thought that `jib.from.image` was a way to specify the base image. Does this override https://github.com/DataBiosphere/terra-billing-profile-manager/blob/main/service/publishing.gradle#L24 ?

Reply:
I think BPM uses the same setup that catalog does, where the gradle config for `jib` handles downloading the cloud profiler and adding it to the base image. The BPM code for this is here: https://github.com/DataBiosphere/terra-billing-profile-manager/blob/main/service/publishing.gradle#L10

Given that setup, I think `jibDockerBuild` would be sufficient by itself.

Reply:
But maybe this form is better than the `jib` method? Or maybe it's doing something different altogether.